lib/join/gitlab/gitlab.go

```diff
@@ -143,3 +164,144 @@ func (c *IDTokenClaims) JoinAttrs() *workloadidentityv1pb.JoinAttrsGitLab {
 
 	return attrs
 }
+
+// CheckIDTokenParams are parameters used to validate GitLab OIDC tokens.
+type CheckIDTokenParams struct {
+	ProvisionToken provision.Token
+	IDToken        []byte
+	Validator      Validator
+}
+
+func (p *CheckIDTokenParams) checkAndSetDefaults() error {
```

Comment:
Suggestion: Since there is no setting of defaults and this looks to only be doing input validation rename the function to indicate as such?
```suggestion
func (p *CheckIDTokenParams) validate() error {
```